backend/executor.go

```diff
@@ -282,11 +282,29 @@ func (g *grpcExecutor) PurgeInstances(ctx context.Context, req *protos.PurgeInst
 	if req.GetPurgeInstanceFilter() != nil {
 		return nil, errors.New("multi-instance purge is not unimplemented")
 	}
+	subOrchestrationInstances := make([]api.InstanceID, 0, 10)
+	id := api.InstanceID(req.GetInstanceId())
+	var err error
+	if req.Recursive {
```

Comment:
Same feedback here as what I mentioned in client.go. Also, is there a way we could put this logic in a single function rather than duplicating it?

Reply:
Done